index.js

```diff
@@ -101,7 +101,7 @@ function asPromise(opts) {
 
 					res.body = data;
 
-					if (opts.json && statusCode !== 204) {
+					if (opts.json && res.body) {
```

Comment:
I'm agree @sindresorhus – if status code is `204` we should not try to parse body, even if there is one.